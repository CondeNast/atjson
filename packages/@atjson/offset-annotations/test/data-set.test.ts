import {
  serialize,
  deserialize,
  Block,
  SliceAnnotation,
} from "@atjson/document";
import OffsetSource, { ColumnType, DataSet, Table } from "../src";

describe("DataSet", () => {
  test("peritext", () => {
    let document = deserialize<Table | DataSet | SliceAnnotation>(
      {
        text: "￼￼column 1 column 2 data 1.1 data 1.2",
        blocks: [
          {
            type: "data-set",
            id: "dataSetId",
            attributes: {
              schema: {
                "column 1": ColumnType.RICH_TEXT,
                "column 2": ColumnType.RICH_TEXT,
              },
              records: [
                {
                  "column 1": {
                    slice: "cell1_1Id",
                    jsonValue: "data 1.1",
                  },
                  "column 2": {
                    slice: "cell1_2Id",
                    jsonValue: "data 1.2",
                  },
                },
              ],
            },
            parents: ["table"],
          },
          {
            type: "table",
            id: "tableId",
            attributes: {
              columns: [
                {
                  name: "column 1",
                  slice: "column1Id",
                  textAlign: "left",
                },
                {
                  name: "column 2",
                  slice: "column2Id",
                  textAlign: "right",
                },
              ],
              dataSet: "dataSetId",
              showColumnHeaders: true,
            },
            parents: [],
          } as Block<Table>,
        ],
        marks: [
          {
            attributes: {
              refs: ["dataSetId"],
            },
            id: "column1Id",
            range: "(2..10]",
            type: "slice",
          },
          {
            attributes: {
              refs: ["dataSetId"],
            },
            id: "column2Id",
            range: "(11..19]",
            type: "slice",
          },
          {
            attributes: {
              refs: ["dataSetId"],
            },
            id: "cell1_1Id",
            range: "(20..28]",
            type: "slice",
          },
          {
            attributes: {
              refs: ["dataSetId"],
            },
            id: "cell1_2Id",
            range: "(29..37]",
            type: "slice",
          },
        ],
      },
      OffsetSource
    );

    expect(serialize(document, { withStableIds: true })).toMatchSnapshot();
  });
});
