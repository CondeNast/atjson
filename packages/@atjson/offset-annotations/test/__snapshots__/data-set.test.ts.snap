// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataSet peritext 1`] = `
{
  "blocks": [
    {
      "attributes": {
        "records": [
          {
            "column 1": {
              "jsonValue": "data 1.1",
              "slice": "M00000002",
            },
            "column 2": {
              "jsonValue": "data 1.2",
              "slice": "M00000003",
            },
          },
        ],
        "schema": {
          "column 1": "rich_text",
          "column 2": "rich_text",
        },
      },
      "id": "B00000000",
      "parents": [],
      "selfClosing": false,
      "type": "data-set",
    },
    {
      "attributes": {
        "columns": [
          {
            "columnName": "column 1",
            "slice": "M00000000",
            "textAlignment": "start",
          },
          {
            "columnName": "column 2",
            "slice": "M00000001",
            "textAlignment": "end",
          },
        ],
        "dataSet": "B00000000",
        "showColumnHeaders": true,
      },
      "id": "B00000001",
      "parents": [],
      "selfClosing": false,
      "type": "table",
    },
  ],
  "marks": [
    {
      "attributes": {
        "refs": [
          "B00000000",
        ],
      },
      "id": "M00000000",
      "range": "(2..10]",
      "type": "slice",
    },
    {
      "attributes": {
        "refs": [
          "B00000000",
        ],
      },
      "id": "M00000001",
      "range": "(11..19]",
      "type": "slice",
    },
    {
      "attributes": {
        "refs": [
          "B00000000",
        ],
      },
      "id": "M00000002",
      "range": "(20..28]",
      "type": "slice",
    },
    {
      "attributes": {
        "refs": [
          "B00000000",
        ],
      },
      "id": "M00000003",
      "range": "(29..37]",
      "type": "slice",
    },
  ],
  "text": "￼￼column 1 column 2 data 1.1 data 1.2",
}
`;
